### ISSUE TITLE
Solving TSP having already found the distances between cities

### ISSUE THREAD

**Issue by xoich** [karepker/little-tsp#3]
Hello, Instead of a map. I would like to use this library to solve directly on the adjacency matrix of the distances. Is it possible?

**Comment by karepker** [comment]
tl;dr Probably, unless your matrix weights are not integers.

To do this, you'd have to make a [`Graph`](https://github.com/karepker/little-tsp/blob/master/src/graph/graph.hpp) object which returns `EdgeCost`s corresponding to your adjacency matrix distances and then pass it to [`FastTspSolver::Solve(...)`](https://github.com/karepker/little-tsp/blob/master/src/tsp_solver/fast.hpp#L11). `Graph` is an abstract class with one concrete implementation that I've written ([`ManhattanGraph`](https://github.com/karepker/little-tsp/blob/master/src/graph/manhattan.hpp)). You can write your own implementation to return weights from your adjacency matrix.

However, because [`EdgeCost` stores only integer values (weights)](https://github.com/karepker/little-tsp/blob/master/src/graph/edge_cost.hpp#L39), you'll need to ensure that your adjacency matrix consists of (and returns) only integer weights.
